#!/bin/sh -e
## no-compress-regexp by Alec Berryman <alec@thened.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Adds several file suffixes not to compress

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
       -patch) patch $patch_opts -p0 < $0;;
       -unpatch) patch $patch_opts -p0 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
--- rdiff_backup/Globals.py.old	2004-09-20 00:12:49.000000000 -0400
+++ rdiff_backup/Globals.py	2004-09-20 00:14:59.000000000 -0400
@@ -166,7 +166,9 @@
 # to .snapshots and .diffs).  The second below will be the
 # compiled version of the first.
 no_compression_regexp_string = "(?i).*\\.(gz|z|bz|bz2|tgz|zip|rpm|deb|" \
-						"jpg|gif|png|jp2|mp3|ogg|avi|wmv|mpeg|mpg|rm|mov|flac|shn)$"
+                               "jpg|jpeg|gif|png|jp2|mp3|ogg|avi|wmv|mpeg|" \
+                               "mpg|rm|mov|flac|shn|pgp|gpg|rz|lzh|zoo|" \
+                               "lharc|rar|arj|asc)$"
 no_compression_regexp = None
 
 # If true, filelists and directory statistics will be split on
--- rdiff-backup.1	2004-09-20 00:18:25.000000000 -0400
+++ new-rdiff-backup.1	2004-09-20 00:38:41.000000000 -0400
@@ -239,8 +239,8 @@
 .TP
 .B "--no-compression-regexp " regexp
 Do not compress increments based on files whose filenames match regexp.
-The default is
-"(?i).*\\.(gz|z|bz|bz2|tgz|zip|rpm|deb|jpg|gif|png|jp2|mp3|ogg|avi|wmv|mpeg|mpg|rm|mov)$"
+The default includes many common audiovisual and archive files, and
+may be found in Globals.py.
 .TP
 .B --no-file-statistics
 This will disable writing to the file_statistics file in the
